manifests/instance.pp

```diff
@@ -228,32 +234,49 @@
       require => Exec["Setup ${title} DS"]
     }
 
-    ensure_resource('service', "dirsrv@${title}",
-      {
-        ensure     => 'running',
-        enable     => true,
-        hasrestart => true
-      }
-    )
+    ensure_resource('ds389::instance::service', $title)
 
-    ds389::config::item { "Set nsslapd-listenhost on ${title}":
-      key          => 'nsslapd-listenhost',
-      value        => $listen_address,
-      admin_dn     => $root_dn,
-      pw_file      => $_ds_pw_file,
-      host         => $listen_address,
-      port         => $port,
-      service_name => $title
+    # This needs to happen first so that we can skip through to ldapi afterwards
+    ds389::instance::attr::set { "Configure LDAPI for ${title}":
+      instance_name => $title,
+      attrs         => {
+        'cn=config' => {
+          'nsslapd-ldapilisten' => 'on',
+          'nsslapd-localssf'    => 99999
+        }
+      },
+      root_dn          => $root_dn,
+      root_pw_file     => $_ds_pw_file,
+      host             => $listen_address,
+      port             => $port,
+      restart_instance => true
     }
 
-    ds389::config::item { "Set nsslapd-securelistenhost on ${title}":
-      key          => 'nsslapd-securelistenhost',
-      value        => $listen_address,
-      admin_dn     => $root_dn,
-      pw_file      => $_ds_pw_file,
-      host         => $listen_address,
-      port         => $port,
-      service_name => $title
+    $_dse_config = {
+      'cn=config' => {
+        'nsslapd-listenhost'       => $listen_address,
+        'nsslapd-securelistenhost' => $listen_address
+      }.merge($general_config)
+    }
+    ds389::instance::attr::set { "Core configuration for ${title}":
+      instance_name => $title,
+      attrs         => $_dse_config,
+      root_dn       => $root_dn,
+      root_pw_file  => $_ds_pw_file,
+      force_ldapi   => true,
+      require       => Ds389::Instance::Attr::Set["Configure LDAPI for ${title}"]
+    }
+
+    if $enable_tls {
```

Comment:
if this instance is disabled, what about the `selinux_port` resource for the secure port in `ds389::instance::tls` ?